core/services/workflows/store/store_db.go

```diff
@@ -70,6 +78,62 @@ type workflowExecutionWithStep struct {
 	WEFinishedAt *time.Time `db:"we_finished_at"`
 }
 
+func (d *DBStore) Start(context.Context) error {
+	return d.StartOnce("DBStore", func() error {
+		d.shutdownWaitGroup.Add(1)
+		go d.pruneDBEntries()
+		return nil
+	})
+}
+
+func (d *DBStore) Close() error {
+	return d.StopOnce("DBStore", func() error {
+		close(d.chStop)
+		d.shutdownWaitGroup.Wait()
+		return nil
+	})
+}
+
+func (d *DBStore) pruneDBEntries() {
+	defer d.shutdownWaitGroup.Done()
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-d.chStop:
+			return
+		case <-ticker.C:
+			ctx1, cancel1 := d.chStop.NewCtx()
+			ctx2, cancel2 := context.WithTimeout(ctx1, 60*time.Second)
```

Comment:
`stopChan.NewCtxWithTimeout` exists I think :)

Reply:
And if you use a closure, then you could do a standard defer :shrug: 
```suggestion
			func() {
				ctx, cancel := d.chStop.CtxWithTimeout(60*time.Second)
				defer cancel
				...
			}()
```